### PR TITLE
2762 feature self serve issue editing edit frontend

### PR DESCRIPTION
**Pull request by captainmangoC4**

### CODE REVIEW

File: netlify/functions/manage-findings.ts [hunk 1]
```diff
@@ -4,57 +4,69 @@ import { Event } from "@netlify/functions/src/function/event";
...
+  sendConfirmationEmail,
+} from "../util/user-utils";
+
+import { apiKey, domain } from "../_config";
 
 async function getFinding(
   username: string,
-  contestId: number,
   issueId: number,
+  contest: Contest
```
Comment (joshuashort) [inline netlify/functions/manage-findings.ts:31]: Converting to an object is probably good, but maybe move it back above issueId?

It's kind of an ordering of contexts.

File: netlify/functions/manage-findings.ts [hunk 2]
```diff
@@ -63,27 +75,22 @@ async function getFinding(
...
+    body: JSON.stringify({ error: "problem fetching finding" }),
   };
 }
 
 async function getFindings(
   username: string,
-  contestId: number,
-  includeTeams: boolean = true
+  includeTeams: boolean = true,
+  contest: Contest
```
Comment (joshuashort) [inline netlify/functions/manage-findings.ts:85]: Same note on context/ordering

File: netlify/functions/manage-findings.ts [hunk 3]
```diff
@@ -63,27 +75,22 @@ async function getFinding(
...
-  // [ ] team findings optional? (query param)
-
-  const contest = await getContest(contestId);
-
-  if (!isContestActive(contest)) {
-    // throw?
-  }
+  // [x] can see specific finding
+  // [x] team findings
+  // [ ] make team findings optional? (query param)
```
Comment (joshuashort) [inline netlify/functions/manage-findings.ts:93]: ?

```suggestion
  // [x] make team findings optional? (query param)
```

File: netlify/functions/manage-findings.ts [hunk 4]
```diff
@@ -123,165 +129,209 @@ async function getFindings(
...
 async function editFinding(
   username: string,
-  contestId: number,
   issueNumber: number,
   data: FindingEditRequest,
-): Promise<Response> {
-  // an authenticated warden can edit a finding
-  //   for active contests
-  //     their own (their teams')
+  contest: Contest
```
Comment (joshuashort) [inline netlify/functions/manage-findings.ts:134]: 😅

File: netlify/functions/submit-finding.ts [hunk 5]
```diff
@@ -306,7 +232,7 @@ exports.handler = async (event) => {
       contest,
       handle: attributedTo,
       address,
-      risk,
+      risk: risk.slice(0, 1),
```
Comment (joshuashort) [inline netlify/functions/submit-finding.ts:235]: todo: explicit mapping

File: src/templates/ReportForm.tsx [hunk 6]
```diff
@@ -77,123 +76,170 @@ const ReportForm = ({ data, location }) => {
...
-            // // will need to reshape the response to be ReportState
-            // // before passing into setting initial state
-            setEndpoint("manage-findings");
-            setIssueId(id);
-            setFindingId(`${contestid}-${id}`);
-
             const q = new URLSearchParams({
-              "contest": contestid,
-              "issue": id,
+              contest: contestid as string,
```
Comment (joshuashort) [inline src/templates/ReportForm.tsx:193]: Below uses toString -- is there a difference?

**Review by joshuashort — APPROVED** [review]
There's a lot going on here, but we've talked about it -- *so*. *good*!

I've left some notes, but none of them are show-stoppers -- minor technical suggestions.

This is certainly mergeable to the feature. Again, very good improvements -- as well as seeming to complete the feature! 🙌

File: netlify/functions/manage-findings.ts (hunk 1 above)
Comment (captainmangoC4) [reply netlify/functions/manage-findings.ts:31]: Oh, sure. Good point.

File: netlify/functions/manage-findings.ts (hunk 3 above)
Comment (captainmangoC4) [reply netlify/functions/manage-findings.ts:93]: Oh, is this actually read from the query param? I thought that part was commented out.

File: netlify/functions/manage-findings.ts (hunk 4 above)
Comment (captainmangoC4) [reply netlify/functions/manage-findings.ts:134]: lol

File: netlify/functions/manage-findings.ts (hunk 3 above)
Comment (captainmangoC4) [reply netlify/functions/manage-findings.ts:93]: I guess in the context of this function, it is optional. The caller just doesn't actually pull the value from query params. I'll add a todo there instead.

File: src/templates/ReportForm.tsx (hunk 6 above)
Comment (captainmangoC4) [reply src/templates/ReportForm.tsx:193]: Good catch. There was a lot of messy type-casting going on around there.

**Pull request merged**